config/secretsmanager/config.go

```diff
@@ -35,13 +35,25 @@ func Configure(p *config.Provider) { //nolint:gocyclo
 				return diff, nil
 			}
 
+			resData, err := schema.InternalMap(r.TerraformResource.SchemaMap()).Data(state, diff)
+			if err != nil {
+				return nil, errors.New("could not construct resource data")
```

Comment:
Is this really something we want to fail the whole observe loop for? I don't fully understand when this could happen, so I don't know how bad it is.

Reply:
under "normal" operation this should never return an error. An error here indicates an issue with the resource schema or the produced state. More specifically, this is a very fundamental operation and the same line is also run by the terraform SDK library already in earlier stages of observe, so any error should be already emitted if any before visiting the custom diff function.So, IMHO it is OK to fail the whole observe here